allen_exporter/exporter.py

```diff
@@ -0,0 +1,416 @@
+        'is_mem_mapped': True,
+        'modality': 'sequence',
+        'n_signals': 1,
+        'n_timestamps': nr_rows, ### lenght of columns
+        'phase_shift_per_signal': False, ### pretty sure it's always false
+        'sampling_rate': float(sampling_rate), ### probably the measurement frequency which is 60 hz, check if we can get this with .metadata
+        'start_time': float(speed_table['timestamps'].iloc[0]) ### first column of timestamps
+    }
+    
+    write_yaml(meta_dict, main_yml)
+
+    treadmill_data = np.column_stack((speed_table['timestamps'],speed_table['speed']))
+    write_mem(treadmill_data, output_dir)
+
+
+
+def dff_export(dff_table, event_table, timestamps, cells_table, depth, output_dir):
+
+    nr_signals = dff_table['dff'].shape[0]
+    d_type = str(dff_table['dff'].iloc[0].dtype)
+    nr_rows = dff_table['dff'].iloc[0].shape[0]
+
+    sampling_rate = (dff_table['dff'].values[0].shape / (timestamps[-1] - timestamps[0]))[0]
+    
+    main_yml = os.path.join(output_dir, "meta.yml")
+    meta_dict = {
+        'dtype': d_type, ### fix later and get dytpe of column 
+        'end_time': float(timestamps[-1]), ### get value of last time
+        'is_mem_mapped': True,
+        'modality': 'sequence',
+        'n_signals': nr_signals,
+        'n_timestamps': nr_rows, ### lenght of columns
+        'phase_shift_per_signal': False, ### pretty sure it's always True here but I can't find any info on how to get the phaseshifts
+        'sampling_rate': float(sampling_rate), ### stated in .metadata this is hardcoded and 
+        'start_time': float(timestamps[0]), ### first column of timestamps
+        'neuron_properties': {
+            'cell_motor_coordinates': 'meta/cell_motor_coordinates.npy',
+            'unit_ids': 'meta/unit_ids.npy',
+            'fields': 'meta/fields.npy'
+        }
+    }
+    
+    write_yaml(meta_dict, main_yml)
+
+    # combine all the rows (individual neurons) into a single matrix
+    dff_values = np.column_stack(dff_table["dff"].values)
+    event_values = np.column_stack(event_table["events"].values)
+
+    write_mem(dff_values, output_dir)
+    write_mem(event_values, output_dir, 'data_events')
+
+    # write additional info into npy files
+    meta_dir = os.path.join(output_dir, "meta/")
+    # cell ids
+    cell_ids = dff_table.index.to_numpy()
+    np.save(meta_dir+"cell_ids.npy", cell_ids)
+    # roi id's
+    roi_id = dff_table["cell_roi_id"].values
+    np.save(meta_dir+"roi_ids.npy", roi_id)
+    # timestamps
+    np.save(meta_dir+"timestamps.npy", timestamps)
+    # coordinates of cells
+    motor_coordinates = cells_table[['x','y']].to_numpy()
+    np.save(meta_dir+"cell_motor_coordinates.npy", motor_coordinates)
+    # ids of cells
+    unit_ids = cells_table.index.to_numpy()
+    np.save(meta_dir+"unit_ids.npy", unit_ids)
+    # ids of cells
+    fields = cells_table['height'].to_numpy()
+    np.save(meta_dir+"fields.npy", fields)
```

Comment:
this is super opinionated so feel free to ignore but I find the version below more concise, hence, easier to read

```suggestion
    meta_dir = os.path.join(output_dir, "meta/")
    np.save(meta_dir + "cell_ids.npy", dff_table.index.to_numpy())
    np.save(meta_dir + "roi_ids.npy", dff_table["cell_roi_id"].values)
    np.save(meta_dir + "timestamps.npy", timestamps)
    np.save(meta_dir + "cell_motor_coordinates.npy", cells_table[['x', 'y']].to_numpy())
    np.save(meta_dir + "unit_ids.npy", cells_table.index.to_numpy())
    np.save(meta_dir + "fields.npy", cells_table['height'].to_numpy())
```

Reply:
Whatever is fine, changed it to your suggestion.